### PR TITLE
Adding live test recordings for new datastore APIs

### PR DESCRIPTION
**Pull request by VamsiSangam**
Command used to record tests - `azdev test VmwareDatastoresScenarioTest --live`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### CODE REVIEW

File: src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+      - application/json
+      ParameterSetName:
+      - --name --resource-group --cluster --private-cloud --nfs-file-path --nfs-provider-ip
+      User-Agent:
+      - AZURECLI/2.19.1 azsdk-python-avsclient/unknown Python/3.8.7 (Windows-10-10.0.18362-SP0)
+    method: PUT
+    uri: https://eastus2euap.management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1?api-version=2021-01-01-preview
+  response:
+    body:
+      string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1","name":"ANFDatastore1","properties":{"netAppVolume":{"nfsFilePath":"ANFVol1FilePath","nfsProviderIp":"10.50.1.4"},"provisioningState":"Pending"},"type":"Microsoft.AVS/privateClouds/clusters/datastores"}'
```
Comment (ctaggart) [inline src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml:118]: `Pending` is not in `DatastoreProvisioningState`.

File: src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+      - keep-alive
+      ParameterSetName:
+      - --name --resource-group --cluster --private-cloud --nfs-file-path --nfs-provider-ip
+      User-Agent:
+      - AZURECLI/2.19.1 azsdk-python-avsclient/unknown Python/3.8.7 (Windows-10-10.0.18362-SP0)
+    method: GET
+    uri: https://eastus2euap.management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1?api-version=2021-01-01-preview
+  response:
+    body:
+      string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1","name":"ANFDatastore1","properties":{"netAppVolume":{"nfsFilePath":"ANFVol1FilePath","nfsProviderIp":"10.50.1.4"},"provisioningState":"Building"},"type":"Microsoft.AVS/privateClouds/clusters/datastores"}'
```
Comment (ctaggart) [inline src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml:210]: `Building` is not in `DatastoreProvisioningState`.

File: src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+      - '0'
+      ParameterSetName:
+      - --name --resource-group --cluster --private-cloud
+      User-Agent:
+      - AZURECLI/2.19.1 azsdk-python-avsclient/unknown Python/3.8.7 (Windows-10-10.0.18362-SP0)
+    method: DELETE
+    uri: https://eastus2euap.management.azure.com/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1?api-version=2021-01-01-preview
+  response:
+    body:
+      string: '{"id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/rasivagu-cloudsan-rg/providers/Microsoft.AVS/privateClouds/rasivagu-sddc/clusters/Cluster-1/datastores/ANFDatastore1","name":"ANFDatastore1","properties":{"netAppVolume":{"nfsFilePath":"ANFVol1FilePath","nfsProviderIp":"10.50.1.4"},"provisioningState":"Building"},"type":"Microsoft.AVS/privateClouds/clusters/datastores"}'
```
Comment (ctaggart) [inline src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml:304]: `Building` is not  in `DatastoreProvisioningState`.

**Review by ctaggart — APPROVED** [review]
These recording are what is needed, thanks! The `provisioningState` values need to be fixed in the service or the spec for `DatastoreProvisioningState`.

File: src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml (hunk 1 above)
Comment (VamsiSangam) [reply src/vmware/azext_vmware/tests/latest/recordings/test_vmware_datastores.yaml:118]: Sure Cameron, I'll investigate this.

**Pull request merged**